src/components/ui/Suggest.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useState } from 'react';
+import nconf from '@qwant/nconf-getter';
+import debounce from 'lodash.debounce';
+
+import SuggestsDropdown from 'src/components/ui/SuggestsDropdown';
+import { suggestResults } from 'src/adapters/suggest_sources';
+import { selectItem, modifyList } from 'src/libs/suggest';
+
+const geocoderConfig = nconf.get().services.geocoder;
+const SUGGEST_MAX_ITEMS = geocoderConfig.maxItems;
+const SUGGEST_USE_FOCUS = geocoderConfig.useFocus;
+const SUGGEST_FOCUS_MIN_ZOOM = 11;
+
+const Suggest = ({ withCategories, withGeoloc }) => {
+  const [items, setItems] = useState([]);
+  const [isOpen, setIsOpen] = useState(false);
+  const [lastQuery, setLastQuery] = useState('');
+  const [isPending, setIsPending] = useState('unset');
+  const [isPendingSubmit, setIsPendingSubmit] = useState(false);
```

Comment:
The point of `isPendingSubmit` in the lifecycle isn't clear to me. Maybe it's a naming problem. If you need to track a kind of status, I'd expect a simple `status` state, with possible values like 'fetching' and 'ready'.
If `isPendingSubmit` is to implement the behavior of submit validation while there is still a query pending (what was called 'preselect' before I think), it could probably be managed in a simpler way by having an async function providing the most up-to-date data, optionnaly waiting for the query result, called when submitting.

Reply:
Great news! now that I re-enabled the submit button, this logic is not needed anymore, we just need to close the dropdown when Enter key is pressed.